src/io/iter_image_recordio.cc

```diff
@@ -212,8 +221,32 @@ inline void ImageRecordIOParser::Init(
   source_.reset(dmlc::InputSplit::Create(
       param_.path_imgrec.c_str(), param_.part_index,
       param_.num_parts, "recordio"));
-  // use 64 MB chunk when possible
-  source_->HintChunkSize(8 << 20UL);
+  if (param_.shuffle_chunk_size > 0) {
+    if (param_.shuffle_chunk_size > 4096) {
+      LOG(INFO) << "Chunk size: " << param_.shuffle_chunk_size
+                 << " MB which is larger than 4096 MB, please set "
+                    "smaller chunk size";
+    }
+    if (param_.shuffle_chunk_size < 4) {
+      LOG(INFO) << "Chunk size: " << param_.shuffle_chunk_size
+                 << " MB which is less than 4 MB, please set "
+                    "larger chunk size";
+    }
+    // 1.1 ratio is for a bit more shuffle parts to avoid boundary issue
+    unsigned num_shuffle_parts =
+        std::ceil(source_->GetTotalSize() * 1.1 /
```

Comment:
for small size, num_shuffle_parts = 1, it uses the split directly, so there is no problem. for large size, the correctness is also ok, but too many chunks may slow down the data loading.